src/main/windows/windowManager.ts

```diff
@@ -464,111 +401,40 @@ export class WindowManager {
 
     restoreMain = () => {
         log.info('restoreMain');
-        if (!this.mainWindow) {
+        if (!MainWindow.get()) {
             this.showMainWindow();
         }
-        if (!this.mainWindow!.isVisible() || this.mainWindow!.isMinimized()) {
-            if (this.mainWindow!.isMinimized()) {
-                this.mainWindow!.restore();
+        const mainWindow = MainWindow.get();
+        if (!mainWindow) {
+            throw new Error('Main window does not exist');
+        }
+        if (!mainWindow.isVisible() || mainWindow.isMinimized()) {
+            if (mainWindow.isMinimized()) {
+                mainWindow.restore();
             } else {
-                this.mainWindow!.show();
+                mainWindow.show();
             }
-            if (this.settingsWindow) {
-                this.settingsWindow.focus();
+            const settingsWindow = SettingsWindow.get();
+            if (settingsWindow) {
+                settingsWindow.focus();
             } else {
-                this.mainWindow!.focus();
+                mainWindow.focus();
             }
-        } else if (this.settingsWindow) {
-            this.settingsWindow.focus();
+        } else if (SettingsWindow.get()) {
+            SettingsWindow.get()?.focus();
```

Comment:
interesting that it asks you to use the optional chaining here, but hey it is what it is, I rather have it optional than with the unwrapping